webapp/api.py

```diff
@@ -1,36 +1,113 @@
-from canonicalwebteam.http import CachedSession
+class CertificationAPI:
+    """
+    Method names and properties to describe and map directly
+    onto the Certification API
+    (at the time of writing, this API is available at
+    https://certification.canonical.com/api/v1)
+    """
 
+    def __init__(self, base_url, session):
+        self.base_url = base_url
+        self.session = session
 
-api_session = CachedSession(
-    fallback_cache_duration=300, file_cache_directory=".webcache"
-)
+    def _get(self, path, params={}):
+        # Remove "None" values from params
+        params = {
+            key: value for key, value in params.items() if value is not None
+        }
 
-base_url = "https://certification.ubuntu.com/api/v1"
+        # Get the JSON data
+        response = self.session.get(
+            f"{self.base_url}/{path.strip('/')}/?format=json", params=params
+        )
 
+        # Raise any HTTP errors
+        response.raise_for_status()
 
-def get(endpoint):
-    return api_session.get(f"{base_url}/{endpoint}")
+        return response
 
+    def certifiedmakes(
```

Comment:
Are we not using underscores too match API method names?

Reply:
Yes, I thought it should match the exact endpoint of the API as closely as possible.